### ISSUE TITLE
Reproduce the best trial results found by ray tune, how to control sampling within small decimals?

### ISSUE THREAD

**Issue by guoxuxu** [ray-project/ray#9980]
### What is your question?
I have some questions while I was trying to reproduce the results using the best config parameters found by the ray tune. First of all, I have a hyperparameter that changes with the total number of epochs, denoted as EPOCHS, which corresponds to ```training_iteration``` as in the following code. I noticed that some trials do not run the whole EPOCHS maybe because that trial consistently produced bad results (metric)? 

```
stop_criteria = {
        metric_name: 0.9,
        "training_iteration": EPOCHS
    }
sched=AsyncHyperBandScheduler(time_attr="training_iteration", metric=metric_name)
algo = HyperOptSearch(
        space=search_space,
        max_concurrent=concurrent_trials,
        metric=metric_name,
        mode="max"
    )
analysis = tune.run(
        Train,
        reuse_actors=True,
        verbose=1,
        num_samples=args.num_samples,
        scheduler=sched,
        search_alg=algo,
        resources_per_trial=resources_per_trial,
        stop=stop_criteria,
    )
```
I set ```training_iteration```, same as EPOCHS in ```stop_criteria``` so that all trials run at most EPOCHS trials. I set ```training_iteration``` in ```AsyncHyperBandScheduler``` because it is required. Don't know if setting ```metric``` in ```AsyncHyperBandScheduler``` means whether or not a trial will be stopped is determined by the value of ```metric```?

Sometimes even the best trial does not run all the EPOCHS, as in the following picture
![image](https://user-images.githubusercontent.com/29363464/89640023-43812600-d8e1-11ea-88d5-5e2d1160069e.png)

As in the best trial JSON file, ```iterations_since_restore``` is the same value as ```training_iteration```, both of which do not equal to EPOCHS. When I use the parameters from the best trial, it can not reproduce the same performance. I set the same random seed (random, torch, Cuda and NumPy) at the beginning of the python file as well as in the ```Train``` initialization.

One more question, what does ```reuse_actors``` actually do?

Thanks a lot ^^

*Ray version and other system information (Python version, TensorFlow version, OS):*
Ray 0.8.5
Python 3.7.4
OS Linux

**Comment by guoxuxu** [comment]
I found a difference !  when I saved the best trial in the JSON file using results from ```analysis.dataframe(metric=metric_name, mode='max')```, the learning rate is truncated to 5 decimal places compared with that result from 
```
    best_trial = analysis.get_best_trial(metric_name, 'max', "all")
    best_config = best_trial.config
```
This may be one of the reasons I cannot reproduce the result using the learning rate from the previous one!

Here comes one more question, how can I prevent sampling from such a small learning rate space, can we force it sampling from a learning rate space that steps within, e.g., 2 decimal places?

**Comment by guoxuxu** [comment]
Answer: https://docs.ray.io/en/master/tune/api_docs/suggestion.html
